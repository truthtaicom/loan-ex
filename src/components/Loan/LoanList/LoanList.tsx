import React from 'react';
import LoanItem, { LoanItemProps } from '../LoanItem/LoanItem';
import { List } from 'antd'

export interface LoanListProps {
  data: LoanItemProps[]
}

export default function LoanList(props: any) {
  return (
    <div>
      <List
        loading={props.loading}
        itemLayout="horizontal"
        dataSource={props.data}
        renderItem={(item: LoanItemProps) => <LoanItem {...item} {...props} />}
      />,
    </div>
  )
}