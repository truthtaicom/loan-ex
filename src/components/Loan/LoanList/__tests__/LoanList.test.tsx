import React from 'react';
import { render } from '@testing-library/react';
import LoanList from '../LoanList';
import loanMock from '../../../../services/loan.json'

describe('<LoanList /> spec', () => {
  const payNow = () => jest.fn()
  it('renders LoanList component', () => {
    const component = render(<LoanList {...loanMock[0]} loading payNow={payNow} />);
    expect(component).toMatchSnapshot();
  });
});
