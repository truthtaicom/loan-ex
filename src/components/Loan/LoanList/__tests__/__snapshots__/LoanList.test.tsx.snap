// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoanList /> spec renders LoanList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ant-list ant-list-split ant-list-loading"
        >
          <div
            class="ant-spin-nested-loading"
          >
            <div>
              <div
                class="ant-spin ant-spin-spinning"
              >
                <span
                  class="ant-spin-dot ant-spin-dot-spin"
                >
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                </span>
              </div>
            </div>
            <div
              class="ant-spin-container ant-spin-blur"
            >
              <div
                style="min-height: 53px;"
              />
            </div>
          </div>
        </div>
        ,
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ant-list ant-list-split ant-list-loading"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div>
            <div
              class="ant-spin ant-spin-spinning"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
          <div
            class="ant-spin-container ant-spin-blur"
          >
            <div
              style="min-height: 53px;"
            />
          </div>
        </div>
      </div>
      ,
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
